packages/internal-test-helpers/lib/test-resolver.js

```diff
@@ -0,0 +1,43 @@
+import { compile } from 'ember-template-compiler';
+
+class Resolver {
+  constructor() {
+    this._registered = {};
+    this.constructor.lastInstance = this;
```

Comment:
Ahh, right the resolver itself is not inside the container.

> It just feels ugly to be relying on reaching.deep.into.__private__.stuff to find the instance.

Agreed. This is one of the reasons I called out a few `__container__` things in this PR.

---

Seems good, thanks for explaining!